lib/workingsetcache/cache.go

```diff
@@ -64,8 +68,31 @@ func Load(filePath string, maxBytes int) *Cache {
 	return loadWithExpire(filePath, maxBytes, *cacheExpireDuration)
 }
 
+// loadFromFileOrNew attempts to load a fastcache.Cache from the given file path
+// If loading fails due to an error (e.g. corrupted or unreadable file), the error is logged
+// and a new cache is created with the specified maxBytes size.
+//
+// This improves observability by surfacing cache loading issues that would otherwise be silently ignored.
+func loadFromFileOrNew(filePath string, maxBytes int) *fastcache.Cache {
+	cache, err := fastcache.LoadFromFile(filePath, maxBytes)
+	if err == nil {
+		return cache
+	}
+
+	if errors.Is(err, os.ErrNotExist) {
+		logger.Infof("cache at path %s missing files; init new cache", filePath)
```

Comment:
all cases below contain `init new cache` message. Would it be simpler to log this message unconditionally with `info` level?